codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostRequestEncoder.java

```diff
@@ -648,16 +681,31 @@ public void addBodyHttpData(InterfaceHttpData data) throws ErrorDataEncoderExcep
                 // add mixedmultipart delimiter, mixedmultipart body header and
                 // Data to multipart list
                 internal.addValue("--" + multipartMixedBoundary + "\r\n");
-                // Content-Disposition: attachment; filename="file1.txt"
-                internal.addValue(HttpHeaderNames.CONTENT_DISPOSITION + ": " + HttpHeaderValues.ATTACHMENT + "; "
-                        + HttpHeaderValues.FILENAME + "=\"" + fileUpload.getFilename() + "\"\r\n");
+
+                if (fileUpload.getFilename().isEmpty()) {
+                    // Content-Disposition: attachment
+                    internal.addValue(HttpHeaderNames.CONTENT_DISPOSITION + ": "
+                            + HttpHeaderValues.ATTACHMENT + "\r\n");
+                } else {
+                    // Content-Disposition: attachment; filename="file1.txt"
+                    internal.addValue(HttpHeaderNames.CONTENT_DISPOSITION + ": "
+                            + HttpHeaderValues.ATTACHMENT + "; "
+                            + HttpHeaderValues.FILENAME + "=\"" + fileUpload.getFilename() + "\"\r\n");
+                }
             } else {
                 internal.addValue("--" + multipartDataBoundary + "\r\n");
-                // Content-Disposition: form-data; name="files";
-                // filename="file1.txt"
-                internal.addValue(HttpHeaderNames.CONTENT_DISPOSITION + ": " + HttpHeaderValues.FORM_DATA + "; "
-                        + HttpHeaderValues.NAME + "=\"" + fileUpload.getName() + "\"; "
-                        + HttpHeaderValues.FILENAME + "=\"" + fileUpload.getFilename() + "\"\r\n");
+
+                if (fileUpload.getFilename().isEmpty()) {
+                    // Content-Disposition: form-data; name="files";
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)